src/GMM.jl

```diff
@@ -83,43 +103,67 @@ function gmm(mf::Function, theta::Vector, theta_l::Vector, theta_u::Vector,
     # NOTE: all handling of data happens right here, because we will use _mf
     #       internally from now on.
     _mf(theta) = max_args(mf) == 1 ? mf(theta): mf(theta, data)
-
+    
     mf0        = _mf(theta)
     nobs, nmom = size(mf0)
     npar       = length(theta)
-
+    
     nl         = length(theta_l)
     nu         = length(theta_u)
-
+    
     @assert nl == nu
     @assert npar == nl
     @assert nobs > nmom
     @assert nmom >= npar
-
+    
     ## mf is n x m
-    smf(theta) = reshape(sum(_mf(theta),1), nmom, 1);
-
+    smf(theta) = reshape(sum(_mf(theta),1), nmom, 1);    
     smf!(θ::Vector, gg) = gg[:] = smf(θ)
-
-    Dsmf = ForwardDiff.forwarddiff_jacobian(smf!, Float64,
-                                           fadtype=:dual, n = npar, m = nmom)
-
-    NLPE = GMMNLPE(smf, Dsmf, W)
-
+    
+    Dsmf = ForwardDiff.forwarddiff_jacobian(smf!, Float64, fadtype=:dual,
+                                            n = npar, m = nmom)
+    NLPE = GMMNLPE(_mf, smf, Dsmf, W)    
     m = MathProgSolverInterface.model(solver)
     l = theta_l
     u = theta_u
     lb = Float64[]
     ub = Float64[]
-    MathProgSolverInterface.loadnonlinearproblem!(m, length(theta), 0, l, u, lb, ub, :Min, NLPE)
-
+    MathProgSolverInterface.loadnonlinearproblem!(m, npar, 0, l, u, lb, ub,
+                                                  :Min, NLPE)
     MathProgSolverInterface.setwarmstart!(m, theta)
     MathProgSolverInterface.optimize!(m)
+    r = GMMResult(MathProgSolverInterface.status(m),
+                                    MathProgSolverInterface.getobjval(m),
+                                    MathProgSolverInterface.getsolution(m),
+                                    nmom, npar, nobs)
+    GMMEstimator(NLPE, r)
+end              
+
+status(me::MomentEstimator) = me.r.status
+StatsBase.coef(me::MomentEstimator) = me.r.coef
+objval(me::MomentEstimator) = me.r.objval
+momentfunction(me::MomentEstimator) = me.e.mf(coef(me))
+jacobian(me::MomentEstimator) = me.e.Dmf(coef(me))
+mfvcov(me::MomentEstimator, k::RobustVariance) = vcov(momentfunction(me), k)
```

Comment:
`CovarianceMatrices` doesn't seem to export `RobustVariance`. We could either change that package so it does or define `const RobustVariance = CovarianceMatrices.RobustVariance` at the top of this file. What do you think it best?